src/Mvc/Mvc.Core/src/Properties/AssemblyInfo.cs

```diff
@@ -3,8 +3,10 @@
 
 using System.Runtime.CompilerServices;
 using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.AspNetCore.Mvc.Routing;
 
 [assembly: TypeForwardedTo(typeof(InputFormatterException))]
+[assembly: TypeForwardedTo(typeof(IRouteTemplateProvider))]
```

Comment:
Why are we still forwarding this one?

Reply:
Ahh, for some route properties...... Hmmm

Reply:
I don't see an equivalent for `IRouteTemplateProvider` that's in Routing already. In the case of `IActionHttpMethodProvider`, `IHttpMethodMetadata` already existed. Am I missing the `IRouteTemplateProvider` equivalent?

Or do you want me to create a new interface? That feels a bit redundant considering it would have the same shape.

Reply:
Microsoft.AspNetCore.~~Mvc~~.Routing

Reply:
Discussed this offline. Going to undo this, add `Microsoft.AspNetCore.Routing.IRouteMetadata` with the same shape. `IRouteTemplateProvider` will implement it. The only difference is `Template` will be renamed to `Pattern` and we'll use a default interface implementation to make it work.